pipelines/combine-zarr-sample-genotypes/Snakefile

```diff
@@ -17,32 +17,33 @@ if "contigs" not in config:
 rule all_merge:
   input:
     expand(
-      os.path.join("{sampleset}/callset.zarr/{chrom}/calldata/{field}"),
-      sampleset=[sampleset],
+      os.path.join(config["samplesets_output_dir"], sampleset, "callset.zarr/{chrom}/calldata/{field}"),
       field=config["fields"],
-      chrom=config["contigs"]),
-    os.path.join(sampleset, "merge_genotypes_config.yml")
+      chrom=config["contigs"]
+    ),
+    os.path.join(config["samplesets_output_dir"], sampleset, "merge_genotypes_config.yml")
 
 
+# NH: This is an input to the contamination pipeline. We didn't use this for phase 3 because we had the phase 2 dataset available, which was curated
 rule all_freqs:
   input:
     expand(
-      os.path.join("{sampleset}/allele_frequencies.zarr/{chrom}/AF"),
-      chrom=config["contigs"],
-      sampleset=[sampleset]),
+      os.path.join(config["samplesets_output_dir"], sampleset, "allele_frequencies.zarr/{chrom}/AF"),
+      chrom=config["contigs"]
+    ),
     os.path.join(sampleset, "merge_genotypes_config.yml")
 
 
 rule merge_zarr:
   input:
-    manifest="{sampleset}/manifest"
+    manifest=os.path.join(config["samplesets_output_dir"], sampleset, "manifest")
   output:
-    zarr=directory("{sampleset}/callset.zarr/{chrom}/calldata/{field}"),
-    done=touch("{sampleset}/callset.zarr/{chrom}/calldata/{field}.complete")
+    zarr=directory(os.path.join(config["samplesets_output_dir"], sampleset, "callset.zarr/{chrom}/calldata/{field}")),
+    done=touch(os.path.join(config["samplesets_output_dir"], sampleset, "callset.zarr/{chrom}/calldata/{field}.complete"))
   params:
-    samples=lambda y: os.path.join(y.sampleset, "manifest"),
-    input_pattern=lambda y: os.path.join(config["data_dir"], "{sample}" + config["gt_suffix"]),
-    output=lambda y: os.path.join(y.sampleset, "callset.zarr"),
+    samples=lambda y: os.path.join(config["samplesets_output_dir"], sampleset, "manifest"),
```

Comment:
minor- but we can get rid of `lambda: y` here, we don't use it for anything

Reply:
awesome. Will do